addons/mail/static/src/core/common/thread.js

```diff
@@ -434,7 +434,7 @@ export class Thread extends Component {
         if (!this.props.thread.selfMember) {
             return false;
         }
-        const seenMessage = this.props.thread.selfMember?.seen_message_id;
-        return Boolean((!previousMessage && !seenMessage) || previousMessage?.eq(seenMessage));
+        const readMessage = this.props.thread.selfMember?.new_message_separator;
+        return Boolean((!previousMessage && !readMessage) || previousMessage?.eq(readMessage));
```

Comment:
This becomes much easier if we store current message rather than previous message.

But the logic is flawed by design, if the target message gets removed from DB, it should take the first message that is after, regardless of whether the id matches or not.

Reply:
Or we could keep the same logic but just do id - 1 ?